src/vncproxy.ml

```diff
@@ -121,43 +119,46 @@ let bind_local_port () =
 
 let connect_to_console session_id console =
   let uri = Uri.of_string console.API.console_location in
-  match%lwt Xen_api_lwt_unix.Lwt_unix_IO.open_connection uri with
+  Xen_api_lwt_unix.Lwt_unix_IO.open_connection uri >>= function
   | Ok ((_, ic), (_, oc)) ->
-    let%lwt () = Lwt_io.write oc (Printf.sprintf "CONNECT %s?%s HTTP/1.0\r\n" (Uri.path uri) (Uri.(encoded_of_query (query uri)))) in
-    let%lwt () = Lwt_io.write oc (Printf.sprintf "Cookie: session_id=%s\r\n" (API.Ref.string_of session_id)) in
-    let%lwt () = Lwt_io.write oc "\r\n" in
-    let%lwt () = Lwt_io.flush oc in
-    let%lwt status = Lwt_io.read_line ic in
+    Lwt_io.write oc (Printf.sprintf "CONNECT %s?%s HTTP/1.0\r\n" (Uri.path uri) (Uri.(encoded_of_query (query uri)))) >>= fun () ->
+    Lwt_io.write oc (Printf.sprintf "Cookie: session_id=%s\r\n" (API.Ref.string_of session_id)) >>= fun () ->
+    Lwt_io.write oc "\r\n" >>= fun () ->
+    Lwt_io.flush oc >>= fun ()->
+    Lwt_io.read_line ic >>= fun status ->
     Printf.fprintf stderr "%s\n%!" status;
     let rest = ref [] in
     let finished = ref false in
-    let%lwt () = while%lwt not !finished do
-        let%lwt line = Lwt_io.read_line ic in
-        Printf.fprintf stderr "%s\n%!" line;
-        if line = ""
-        then finished := true
-        else rest := line :: !rest;
-        return ()
-      done in
-    return (ic, oc)
+    let rec loop () =
+      if not !finished then
+        (Lwt_io.read_line ic >>= fun line ->
+         Printf.fprintf stderr "%s\n%!" line;
+         if line = "" then finished := true
+         else rest := line :: !rest;
+         return ()
+
+        ) >>= loop
+      else Lwt.return_unit
+    in
+    loop () >>= fun () -> return (ic, oc)
   | Error e -> fail e
 
 let proxy (a_ic, a_oc) (b_ic, b_oc) =
   let copy x y =
-    try%lwt
-      while%lwt true do
-        let%lwt c = Lwt_io.read_char x in
-        Lwt_io.write_char y c
-      done
-    with End_of_file -> return () in
+    Lwt.catch
+      (fun () ->
+         let rec loop () = (Lwt_io.read_char x >>= fun c -> Lwt_io.write_char y c) >>= loop in
```

Comment:
Just a small stylistic change
```ocaml
let rec loop () =
  Lwt_io.read_char x 
  >>= fun c -> Lwt_io.write_char y c
  >>= loop
in
```

Reply:
Wouldn't this match better the other bindings in the file?
```
let rec loop () =
  Lwt_io.read_char x >>= fun c ->
  Lwt_io.write_char y c >>= loop
in
```

Reply:
Yes, I like your version more